src/client/src/components/TopBar.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { UserToken } from '../../../../types';
+
+export interface TopbarProps {
+    user: UserToken | null;
+    setUser: React.Dispatch<React.SetStateAction<UserToken | null>>;
+}
+
+export const Topbar = (props: TopbarProps): JSX.Element => {
+    const navigate = useNavigate();
+    //The username does not rerender when logging in. Need to fix
+    const user = props.user;
+
+    const logOut = () => {
+        window.localStorage.removeItem('loggedGraphUser');
+        props.setUser(null);
+        navigate('/user/login');
+    };
+    return (
+        <div className="topbar">
```

Comment:
I'd do two separate `div`s for both cases here (so `if (user) { return <div>... } else { return <div>...}`